app/ui/tasks/TaskCardComponent.tsx

```diff
@@ -0,0 +1,105 @@
+"use client";
+
+import { statusColor, typeColor } from "@/app/helpers/badgeColors";
+import { formatDate } from "@/app/helpers/dateFormat";
+import { getTask } from "@/app/seed/route";
+import Loading from "@/app/task-managment/loading";
+import { ITask, TaskStatuses } from "@/app/types/models";
+import { ChevronDownIcon } from "@chakra-ui/icons";
+import {
+  Text,
+  Heading,
+  Stack,
+  Box,
+  Divider,
+  Spacer,
+  Badge,
+  Menu,
+  MenuButton,
+  MenuList,
+  MenuItem,
+  Button,
+} from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import TaskEditComponent from "./TaskEditComponent";
+
+export default function TaskCardComponent({ id }: { id: string }) {
```

Comment:
I suggest re-writing this in "a NextJs way". I think that most of the page can be on server side.
Use server actions for forms. Also, this will make your work with forms easier https://react-hook-form.com/